src/common.cpp

```diff
@@ -1849,61 +1856,122 @@ bool valid_func_name(const wcstring &str) {
 
 /// Return the path to the current executable. This needs to be realpath'd.
 std::string get_executable_path(const char *argv0) {
+    static std::string result;
+    if (!result.empty()) return result;
     char buff[PATH_MAX];
-
+    std::string buffstr;
```

Comment:
Please define this where it is used. This probably means defining it *before* that one if-block where it first appears (instead of inside it).

Reply:
I put it at the top in front of all the platform checks because it is used by all platforms iirc.

Reply:
It's not used by the Apple path *or* the KERN_PROC_PATHNAME one.

It appears the first use is in the /proc bit, followed by the $PATH/$PWD bit.

Reply:
I remembered wrong. I'm working on multiple things right now otherwise I probably would've noticed that.